src/develop/develop.c

```diff
@@ -2436,13 +2441,23 @@ int dt_dev_distort_transform_plus(dt_develop_t *dev, dt_dev_pixelpipe_t *pipe, c
     modules = g_list_next(modules);
     pieces = g_list_next(pieces);
   }
+  if (transf_direction == DT_DEV_TRANSFORM_DIR_ALL 
```

Comment:
Should we skip this if preview_downsampling == 1 to speed up this case when there is many points.

Reply:
Nice idea, done.